crypto/rand/randfile.c

```diff
@@ -55,202 +55,15 @@
  */
 int RAND_load_file(const char *file, long bytes)
 {
-    /*
-     * The load buffer size exceeds the chunk size by the comfortable amount
-     * of 'RAND_DRBG_STRENGTH' bytes (not bits!). This is done on purpose
-     * to avoid calling RAND_add() with a small final chunk. Instead, such
-     * a small final chunk will be added together with the previous chunk
-     * (unless it's the only one).
-     */
-#define RAND_LOAD_BUF_SIZE (RAND_BUF_SIZE + RAND_DRBG_STRENGTH)
-    unsigned char buf[RAND_LOAD_BUF_SIZE];
-
-#ifndef OPENSSL_NO_POSIX_IO
-    struct stat sb;
-#endif
-    int i, n, ret = 0;
-    FILE *in;
-
-    if (bytes == 0)
-        return 0;
-
-    if ((in = openssl_fopen(file, "rb")) == NULL) {
-        ERR_raise_data(ERR_LIB_RAND, RAND_R_CANNOT_OPEN_FILE,
-                       "Filename=%s", file);
-        return -1;
-    }
-
-#ifndef OPENSSL_NO_POSIX_IO
-    if (fstat(fileno(in), &sb) < 0) {
-        ERR_raise_data(ERR_LIB_RAND, RAND_R_INTERNAL_ERROR,
-                       "Filename=%s", file);
-        fclose(in);
-        return -1;
-    }
-
-    if (bytes < 0) {
-        if (S_ISREG(sb.st_mode))
-            bytes = sb.st_size;
-        else
-            bytes = RAND_DRBG_STRENGTH;
-    }
-#endif
-    /*
-     * Don't buffer, because even if |file| is regular file, we have
-     * no control over the buffer, so why would we want a copy of its
-     * contents lying around?
-     */
-    setbuf(in, NULL);
-
-    for (;;) {
-        if (bytes > 0)
-            n = (bytes <= RAND_LOAD_BUF_SIZE) ? (int)bytes : RAND_BUF_SIZE;
-        else
-            n = RAND_LOAD_BUF_SIZE;
-        i = fread(buf, 1, n, in);
-#ifdef EINTR
-        if (ferror(in) && errno == EINTR) {
-            clearerr(in);
-            if (i == 0)
-                continue;
-        }
-#endif
-        if (i == 0)
-            break;
-
-        RAND_add(buf, i, (double)i);
-        ret += i;
-
-        /* If given a bytecount, and we did it, break. */
-        if (bytes > 0 && (bytes -= i) <= 0)
-            break;
-    }
-
-    OPENSSL_cleanse(buf, sizeof(buf));
-    fclose(in);
-    if (!RAND_status()) {
-        ERR_raise_data(ERR_LIB_RAND, RAND_R_RESEED_ERROR, "Filename=%s", file);
-        return -1;
-    }
-
-    return ret;
+    return -1;
```

Comment:
Making this fail like this will break existing applications calling it, of which there are quite a few. Granted, many of them error check incorrectly `if (!RAND_load_file(...))` and will still succeed.

We made this return `bytes` unless it's `-1` in which case we return `123456` (any positive value will do). In other words, we pretend we have added `bytes` or `the whole file` to the 'entropy pool'.

Reply:
Thanks I fixed the retval to be like what you said.  Ready for another look.